presto-main/src/main/java/io/prestosql/execution/buffer/PageCodecMarker.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution.buffer;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Encodes boolean properties for {@link SerializedPage} by using a bitmasking strategy, allowing
+ * up to 8 such properties to be stored in a single byte
+ */
+public enum PageCodecMarker
+{
+    COMPRESSED(1),
+    ENCRYPTED(2);
+
+    private final int mask;
+
+    PageCodecMarker(int bit)
+    {
+        checkArgument(bit > 0 && bit <= 8, "PageCodecMarker bit must be between 1 and 8. Found: %s", bit);
+        this.mask = (1 << (bit - 1));
+    }
+
+    public boolean isSet(byte value)
```

Comment:
Instead of exposing the bit orientedness of this to the callers, I suggest we use a `Set< PageCodecMarker>` in the main code, and then we in `SerializedPage` we convert to a `byte` for memory efficiency.

Reply:
I think there are some tricky memory overhead considerations of going with a full Set<PageCodecMarker>. I think I can come up with a happy medium of making the interface a little less awkward by implementing a PageCodecMarkerSet class that works a little bit like a BitSet over the underlying byte so we don’t incur the per Page overhead associated with Set (even EnumSet) instances. I’ll put that together and see what you think.

Reply:
I'm still in the process of cleaning up the code based on other comments, but [here's a revised first commit](https://github.com/pettyjamesm/prestosql/commit/8ccb417f8c57010e030615a34a309b897228c2b9) with the bitset-like API we talked about. Let me know if you think it's an improvement over the raw byte based API (note: don't review it for anything other than style of API, I haven't even tried to compile it yet). IMO- it still creates some awkwardness where the choice is either to still expose the `PageCodecMarker#isSet(byteValue)` or to force the code to box the `byte` into a `MarkerSet` just to do a `MarkerSet#contains(PageCodecMarker)` check. I chose to leave that API public in the commit for that reason.